routes/index.js

```diff
@@ -1,31 +1,66 @@
 const apiRouter = require('express').Router();
-const { getAllProducts, getProductById } = require('../db');
+const jwt = require('jsonwebtoken')
+const JWT_SECRET = 'codebrigaderules'
 
-apiRouter.get("/", (req, res, next) => {
+const { getUserById } = require('../db');
+
+apiRouter.get("/", (req, res) => {
   res.send({
     message: "API is under construction!"
   });
 });
 
-apiRouter.get('/products', async (req, res, next) => {
-  try {
-      res.setHeader('Access-Control-Allow-Origin', 'http://localhost:3000');
-      const products = await getAllProducts();
-      res.send(products);
-  } catch (error) {
-      next(error);
+apiRouter.get('/masks', (req, res, next) => {
+  res.send({message: 'All is safe'});
+});
+
+apiRouter.use(async (req, res, next) => {
+  const prefix = 'Bearer ';
+  const auth = req.header('Authorization');
+
+  if (!auth) { 
+  next();
+  } else if (auth.startsWith(prefix)) {
+    const token = auth.slice(prefix.length);
+    try {
+    const { id } = jwt.verify(token, JWT_SECRET);
+
+    if (id) {
+      req.user = await getUserById(id);
```

Comment:
This might automatically error... but it might be best to say `if(!user) throw 'no user by that id'`